input-output-hk/etc-client
# [ETCM-491] Extend generated bash script with absolute path configs

# Description

See https://github.com/input-output-hk/mantis/issues/851

# Proposed Solution

As @aakoshh commented in the issue, one solution is to extend the the generated bash script file so that we can set the config files paths correctly.

# Testing

These instructions should work for Unix and for Windows (just by changing `/` to `\`).

1. Build: `sbt dist`
2. Move to `target/universal`: `cd target/universal`
3. Unzip: `unzip mantis-3.2.1.zip`
4. Verify the following commands

### Should use etc by default (this is the test for the original issue)

- [x] `mantis-3.2.1/bin/mantis` - Check that it starts the client and connects to etc network. 

### Should use given chain

- [x] `mantis-3.2.1/bin/mantis-launcher sagano` - Check that it starts and connects to sagano testnet.

### Should not break other scripts
All of the following should be valid and work as usual
- [x] `mantis-3.2.1/bin/mantis-vm 8080`
- [x] `mantis-3.2.1/bin/eckeygen 8`
- [x] `mantis-3.2.1/bin/signatureValidator f0b91b55bdea6a6da2bcbd1270451e9ca014747b97e190a5a4275bad8eeda7436ab8081f0df5741a86a1bfb62798221103fff5898c4f768b707ff34b110329ed 6bf862422f0103d8ca6847019c0113c9b343a2489c2dc3fc67d853e87d98f68e5a5be1d021ec706197960912037d46bde44c3534a4ff562fd48e664f32c1392d1c 5fe7f977e71dba2ea1a68e21057beebb9be2ac30c6410aa38d4f3fbe41dcffd2`
- [x] `mantis-3.2.1/bin/faucet-server`
